src/client/datascience/notebook/outputs/plotSaveHandler.ts

```diff
@@ -0,0 +1,107 @@
+import { inject, injectable } from 'inversify';
+import { NotebookCellOutput, NotebookEditor, Uri } from 'vscode';
+import { IApplicationShell, IWorkspaceService } from '../../../common/application/types';
+import { IFileSystem } from '../../../common/platform/types';
+import { swallowExceptions } from '../../../common/utils/decorators';
+import { DataScience } from '../../../common/utils/localize';
+import * as path from 'path';
+import { saveSvgToPdf } from '../../plotting/plotViewer';
+
+const svgMimeType = 'image/svg+xml';
+const imageExtensionForMimeType: Record<string, string> = {
+    'image/png': 'png',
+    'image/jpeg': 'jpeg',
+    'image/gif': 'gif',
+    [svgMimeType]: 'svg'
+};
+
+@injectable()
+export class PlotSaveHandler {
+    constructor(
+        @inject(IApplicationShell) private readonly shell: IApplicationShell,
+        @inject(IFileSystem) private readonly fs: IFileSystem,
+        @inject(IWorkspaceService) private readonly workspace: IWorkspaceService
+    ) {}
+
+    @swallowExceptions()
+    public async savePlot(editor: NotebookEditor, outputId: string, mimeType: string) {
+        if (editor.document.isClosed) {
+            return;
+        }
+        const output = getOutputItem(editor, outputId, mimeType);
+        if (!output) {
+            throw new Error(`Nolot to save ${editor.document.uri.toString()}, id: ${outputId} for ${mimeType}`);
```

Comment:
Is it intentional to both @swallowExceptions and throw Errors here? Isn't that just traceError in the end?

Reply:
Lol yes, initially I wrote throw, but I didn't want the calling code to care about the exceptions, then added swallow..
Will change to just trace, that's simpler.

Reply:
Done